polkadot/xcm/src/v3/multilocation.rs

```diff
@@ -444,6 +444,20 @@ impl MultiLocation {
 			}
 		}
 	}
+
+	/// Return the MultiLocation subsection identifying the chain that `self` points to.
+	pub fn chain_location(mut self) -> MultiLocation {
+		// start popping junctions until we reach chain identifier
+		while let Some(j) = self.last() {
+			if matches!(j, Junction::Parachain(_) | Junction::GlobalConsensus(_)) {
+				// return chain subsection
+				return self
+			} else {
+				(self, _) = self.split_last_interior();
```

Comment:
I'm not sure if this is a good idea to mutate `self` just to get the `GlobalConsensus` junction -- it seems to me that the main purpose is just to acquire information, not mutate the data structure. In other words, this function should take a `&self` and disable mutability.

Reply:
done